src/ol/renderer/canvas/VectorImageLayer.js

```diff
@@ -103,9 +103,13 @@ class CanvasVectorImageLayerRenderer extends CanvasImageLayerRenderer {
       });
 
       listen(image, EventType.CHANGE, function() {
+        /**
+         * @type {CanvasVectorImageLayerRenderer}
+         */
+        const this_ = this;
```

Comment:
instead of this, we could use `bind(this)`:
```js
listen(image, EventType.CHANGE, function() {
  if (image.getState() === ImageState.LOADED) {
    this.image_ = image;
    this.skippedFeatures_ = skippedFeatures;
  }
}.bind(this);
```
I think it's prettier than creating this `this_` variable

Reply:
We can also use an @this to type the this...

Reply:
I've tried but typescript did not seem to take this into account.